src/main/java/org/apache/commons/lang3/ObjectUtils.java

```diff
@@ -1051,6 +1052,10 @@ public static boolean isEmpty(final Object object) {
         if (object instanceof Map<?, ?>) {
             return ((Map<?, ?>) object).isEmpty();
         }
+        if (object instanceof Optional<?>) {
+        	return !((Optional<?>) object).isPresent();
+        }
+        
```

Comment:
What if it's an Optional with an empty string in it?

Reply:
The optional would be non-empty. Take the following code snippet:
```java
Optional<String> optional = Optional.of("");
boolean empty = ObjectUtils.isEmpty(optional);
// empty is false
```
In the case of Optionals, we're concerned whether the optional is empty, not if the contents of the optional is empty.

This is similar to having an array or collection of length 1 that only contains an empty string. In that case, ObjectUtils.isEmpty would still return false.

I can update the JavaDoc to make it more clear.

(And unrelated to this conversation, I notices I accidently slipped a couple tabs instead of spaces in. I'll fix that shortly.)